bemani/data/mysql/music.py

```diff
@@ -105,8 +114,37 @@ def __get_musicid(self, game: GameConstants, version: int, songid: int, songchar
             },
         )
         if cursor.rowcount != 1:
-            # music doesn't exist
-            raise Exception(f"Song {songid} chart {songchart} doesn't exist for game {game} version {version}")
+            if game == GameConstants.DDR and version == VersionConstants.DDR_SUPERNOVA_2:
```

Comment:
Not sure of a clean way to extract the song list from the game yet, so making them on demand here. There is probably a way to do it but the files on the hdd are encrypted so need to think of a method of extracting them.

Reply:
You'll need to figure that out, or provide a song DB .tsv or something, because on-demand music stuff isn't supposed to be here. That's why this module has no way of getting the next ID. From it's perspective, the music DB is read-only.

I assume this also means no front-end? I think you mentioned that.

Reply:
Yeah currently no front-end, but it's something I want to add so that the events can be managed among other things.

I wasn't aware of the .tsv files, this might be the approach I'll have to take here, there isn't any public tools for decrypting the dnas blobs that I know of and the easiest way is most likely to extract the data from memory once loaded.